### PR TITLE
NEXUS-21749 Resolve technical debt in Helm

### PR DESCRIPTION
**Pull request by proteus1121**
https://issues.sonatype.org/browse/NEXUS-21749

### CODE REVIEW

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmFacet.java [hunk 1]
```diff
@@ -12,36 +12,63 @@
...
 public interface HelmFacet
     extends Facet
 {
-  Asset findOrCreateAssetWithComponent(
-      final String assetPath,
-      final AssetKind assetKind,
-      final StorageTx tx,
-      final Bucket bucket,
-      final HelmAttributes chart);
+  List<HashAlgorithm> HASH_ALGORITHMS = ImmutableList.of(SHA1, SHA256);
```
Comment (chernikovmaksym) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmFacet.java]: We don't need this in interface. It's a part of implementation. It's better to put it into `Util` class

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/HelmFacetImpl.java [hunk 2]
```diff
@@ -37,73 +61,157 @@
...
+   * Find an asset by its name.
+   *
+   * @return found Optional<Asset> or Optional.empty if not found
+   */
+  @Override
+  public Optional<Asset> findAsset(final String assetName) {
+    StorageTx tx = UnitOfWork.currentTx();
+    Bucket bucket = tx.findBucket(getRepository());
+    Asset asset = tx.findAssetWithProperty(P_NAME, assetName, bucket);
+    return asset != null ? Optional.of(asset) : Optional.empty();
```
Comment (chernikovmaksym) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/HelmFacetImpl.java]: `Optional.ofNullable()`?

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/HelmFacetImpl.java [hunk 3]
```diff
@@ -37,73 +61,157 @@
...
-      Component component = findOrCreateComponent(tx, bucket, chart);
-      asset = tx.createAsset(bucket, component);
-      asset.name(assetPath);
-      asset.formatAttributes().set(P_ASSET_KIND, assetKind.name());
-    }
-
-    helmAssetAttributePopulator.populate(asset.formatAttributes(), chart);
+    Optional<Asset> assetOpt = findAsset(assetPath);
+    Asset asset = assetOpt.orElseGet(() ->
+        isComponentRequired ? createAsset(assetPath, assetKind, (StorageTx tx, Bucket bucket) -> tx.createAsset(bucket, findOrCreateComponent(helmAttributes)))
```
Comment (chernikovmaksym) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/HelmFacetImpl.java]: Line is too long and it's quite unreadable. Try reformatting

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/hosted/HelmHostedFacetImpl.java [hunk 4]
```diff
@@ -83,15 +78,17 @@ public Content get(final String path) {
...
       return null;
     }
+
+    Asset asset = assetOpt.get();
     if (asset.markAsDownloaded()) {
       tx.saveAsset(asset);
     }
 
-    return helmDataAccess.toContent(asset, tx.requireBlob(asset.requireBlobRef()));
+    return helmFacet.toContent(assetOpt.get(), tx.requireBlob(asset.requireBlobRef()));
```
Comment (chernikovmaksym) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/hosted/HelmHostedFacetImpl.java]: `assetOpt.get()` -> `asset`

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmFacet.java (hunk 1 above)
Comment (mlukaretkyi) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmFacet.java]: yeap, it is a good point

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/HelmFacetImpl.java (hunk 2 above)
Comment (proteus1121) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/HelmFacetImpl.java]: Done

**Pull request closed without merging**